packages/playground/src/components/dashboard/portal/NodeMintingDetails.vue

```diff
@@ -0,0 +1,138 @@
+<template>
+  <v-container>
+    <v-row>
+      <v-col cols="3">
+        <v-menu ref="menu" v-model="showMenu" :close-on-content-click="false" transition="scale-transition">
+          <template v-slot:activator="{ props }">
+            <v-text-field v-bind="props" label="Select Month and Year" readonly :model-value="selectedData" />
+          </template>
+
+          <v-card>
+            <v-card-text class="d-flex">
+              <v-select v-model="selectedMonth" :items="months" label="Month"></v-select>
+
+              <v-select v-model="selectedYear" :items="years" label="Year"></v-select>
+            </v-card-text>
+          </v-card>
+        </v-menu>
+      </v-col>
+      <v-col cols="9">
+        <v-card outlined>
+          <v-card-title>
+            <span class="headline">Minting Details</span>
+          </v-card-title>
+          <div class="receipt-body" v-if="receipts?.length">
+            <v-card outlined v-for="receipt in receipts" :key="receipt.hash">
+              <v-container>
+                <v-row>
+                  <v-col cols="10">
+                    <v-card-text>
+                      <span class="font-weight-bold">Date of Payout :</span>
+                      {{ getDateFromTimestamp(receipt.endPeriodTimestamp) }}
+                      <br />
+                      <span class="font-weight-bold">Node TFT Amount :</span> {{ receipt.tft || 0 }} TFT
+                      <span v-if="receipt.fixupReward">+ {{ receipt.fixupReward || 0 }} TFT FixedUp</span>
+                      <br />
+                      <span class="font-weight-bold">Cloud Units :</span>
+                      <ul>
+                        <li inset v-for="(val, key) in receipt.cloud_units" :key="key">
+                          <b style="text-transform: uppercase">{{ key }}:</b> {{ val }}
+                        </li>
+                      </ul>
+                    </v-card-text>
+                  </v-col>
+                  <v-col cols="2">
+                    <v-chip class="ma-4 me-auto" small :color="getChipColor(receipt.type)">{{ receipt.type }} </v-chip>
+                  </v-col>
+                </v-row>
+              </v-container>
+            </v-card>
+          </div>
+          <div v-else>
+            <v-card-text class="font-weight-bold">No receipts found for this month</v-card-text>
+          </div>
+          <v-card-actions>
+            <v-spacer></v-spacer>
+            <v-btn color="blue" @click="downloadNodeReceipt" :disabled="!node.receipts">Download Node Receipt</v-btn>
+          </v-card-actions>
+        </v-card>
+      </v-col>
+    </v-row>
+  </v-container>
+</template>
+
+<script lang="ts">
+import { jsPDF } from "jspdf";
+import moment from "moment";
+import { computed, ref } from "vue";
+
+import { generateReceipt, type receiptInterface } from "@/utils/node";
+
+export default {
+  name: "NodeMintingDetails",
+  props: ["node"],
```

Comment:
This can only be done in a setup script. The only way to define props in the current script is by defining them in an array of strings as displayed above.

Reply:
In the current script, the only way to define a prop is by adding it as a string to props array. Other than that, I'd have to use setup script which will create several changes to the code. check the [docs](https://vuejs.org/guide/components/props.html#props-declaration).